mesonpy/__init__.py

```diff
@@ -951,6 +952,27 @@ def _project(config_settings: Optional[Dict[Any, Any]]) -> Iterator[Project]:
         yield project
 
 
+def _get_ninja_if_needed() -> List[str]:
+    # only add ninja if it is not present or is too low a version
+    ninja = shutil.which('ninja')
+    if ninja is None:
+        return []
+
+    result = subprocess.run(['ninja', '--version'], check=False, text=True, capture_output=True)
```

Comment:
This should use the returned value of shutil.which above.

Reply:
It doesn't matter if we are only looking for one command. But when we look for three, yes, this would be better.